main.go

```diff
@@ -995,7 +999,11 @@ func main() {
 		if s, err := goenv.GorootVersionString(goenv.Get("GOROOT")); err == nil {
 			goversion = s
 		}
-		fmt.Printf("tinygo version %s %s/%s (using go version %s and LLVM version %s)\n", goenv.Version, runtime.GOOS, runtime.GOARCH, goversion, llvm.Version)
+		if !strings.HasSuffix(goenv.Version, "-dev") {
+			fmt.Printf("tinygo version %s %s/%s (using go version %s and LLVM version %s)\n", goenv.Version, runtime.GOOS, runtime.GOARCH, goversion, llvm.Version)
+		} else {
+			fmt.Printf("tinygo version %s-%s %s/%s (using go version %s and LLVM version %s)\n", goenv.Version, gitSha1, runtime.GOOS, runtime.GOARCH, goversion, llvm.Version)
+		}
```

Comment:
You could keep a single `Printf` line here, reducing duplication:

```suggestion
		version := goenv.Version
		if !strings.HasSuffix(goenv.Version, "-dev") && gitSha1 != "" {
			version += "-" + gitSha1
		}
		fmt.Printf("tinygo version %s %s/%s (using go version %s and LLVM version %s)\n", version, runtime.GOOS, runtime.GOARCH, goversion, llvm.Version)
```

Also, you can build TinyGo without going through the Makefile so there needs to be a check that `gitSha1` is set.

Reply:
The less duplication, the better.